aequilibrae/paths/results/assignment_results.py

```diff
@@ -104,11 +102,16 @@ def reset(self) -> None:
         if self.link_loads is not None:
             self.no_path.fill(0)
             self.link_loads.fill(0)
-            self.total_link_loads.fill(0)
+            self.link_loads.fill(0)
+            self.compact_link_loads.fill(0)
+            self.compact_total_link_loads.fill(0)
         else:
             raise ValueError("Exception: Assignment results object was not yet prepared/initialized")
 
     def __redim(self):
+        self.compact_link_loads = np.zeros((self.compact_links + 1, self.classes["number"]), self.__float_type)
+        self.compact_total_link_loads = np.zeros(self.compact_links, self.__float_type)
```

Comment:
Yes.  It happens that some links get completely removed when the network is simplified (due to coding error, a link segment on a one-way road has the wrong direction), so instead of having to deal with this through conditions, I put one extra element in the compact link loads and refer the dropped links to it.
This ends up being the case for links only available for some classes and not others, which appear with a_node==b_node in the uncompressed graph but are completely removed from the simplified one.